XamarinCommunityToolkit/Core/FileMediaSource.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Core
+{
+	[TypeConverter(typeof(FileMediaSourceConverter))]
+	public sealed class FileMediaSource : MediaSource
+	{
+		public static readonly BindableProperty FileProperty = BindableProperty.Create(nameof(File), typeof(string), typeof(FileMediaSource), default(string));
+
+		public string File
+		{
+			get => (string)GetValue(FileProperty);
+			set => SetValue(FileProperty, value);
+		}
+
+		public override string ToString() => $"File: {File}";
+
+		public static implicit operator FileMediaSource(string file) => (FileMediaSource)FromFile(file);
+
+		public static implicit operator string(FileMediaSource file) => file?.File;
+
+		protected override void OnPropertyChanged(string propertyName = null)
```

Comment:
How about passing propertyChanged delegate to bindable property instead of overriding this method?)

Reply:
@AndreiMisiukevich  I think that is impossible, the `FileMediaSource` doesn't know which type the `BindableObject` will be.

Reply:
Why? It will be **FileMediaSource** because this bindable property belongs to **FileMediaSource**

Reply:
Could you pls address this one as well?